provisioning/baremetal_secrets.go

```diff
@@ -62,6 +64,16 @@ func CreateMariadbPasswordSecret(client coreclientv1.SecretsGetter, targetNamesp
 			ObjectMeta: metav1.ObjectMeta{
 				Name:      baremetalSecretName,
 				Namespace: targetNamespace,
+				OwnerReferences: []metav1.OwnerReference{
```

Comment:
Is there any reason not to use SetControllerReference here, too? I guess that would mean moving this Secret to a variable, but that seems OK.

Reply:
it just means passing in r.Schema as well. it just felt messy, i can change it and see what you think